server/src/test/java/org/opensearch/action/search/AbstractSearchAsyncActionTests.java

```diff
@@ -313,6 +333,60 @@ public void testSendSearchResponseDisallowPartialFailures() {
         assertEquals(requestIds, releasedContexts);
     }
 
+    public void testOnPhaseListenersFailure() {
+        SearchCoordinatorStatsTesting searchCoordinatorStatsTesting = new SearchCoordinatorStatsTesting();
+        SearchRequestOperationsListener testListener = createSearchRequestOperationsListener(searchCoordinatorStatsTesting);
+        final List<SearchRequestOperationsListener> requestOperationListeners = new ArrayList<>(Arrays.asList(testListener, testListener));
+
+        SearchQueryThenFetchAsyncAction action = createSearchQueryThenFetchAsyncAction();
+        action.setSearchListenerList(requestOperationListeners);
+        action.setCurrentPhase(action);
+        action.onPhaseFailure(new SearchPhase("test") {
+            @Override
+            public void run() {
+
+            }
+        }, "message", null);
+        assertEquals(2, searchCoordinatorStatsTesting.queryPhaseFailure.get());
```

Comment:
These doesn't make sense. Try to use make original stats like metrics, verify time taken, query count etc. You can add a deliberate delay for each phase, and verify whether actual time taken matches with it. Similar for count metrics.